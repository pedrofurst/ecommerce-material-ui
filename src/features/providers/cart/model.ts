import { ProductType } from '../product/model';

export type CartContextType = {
  cart: CartType;
  addToCart: (product: ProductType) => void;
  removeFromCart: (product: ProductType) => void;
  clear: () => void;
};

export type CartType = {
  products: ProductType[];
};
