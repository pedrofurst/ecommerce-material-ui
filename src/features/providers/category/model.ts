export type CategoryContextType = {
  categories: string[];
  updateCategories: (categories: string[]) => void;
  selectedCategory: string;
  updateSelectedCategory: (category: string) => void;
};
