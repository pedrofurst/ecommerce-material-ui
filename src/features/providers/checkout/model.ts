export type CheckoutContextType = {
  shippingAddress: ShippingAddressType;
  creditCard: CreditCardType;
  updateShippingAddress: (value: ShippingAddressType) => void;
  updateCreditCard: (value: CreditCardType) => void;
};

export type ShippingAddressType = {
  firstName: string;
  lastName: string;
  streetAddress: string;
  city: string;
  postcode: string;
  country: string;
  state: string;
};

export type CreditCardType = {
  name: string;
  number: string;
  cvc: string;
  expiry: string;
};
