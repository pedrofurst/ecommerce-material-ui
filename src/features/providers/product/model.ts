export type ProductContextType = {
  products: ProductType[];
  updateProducts: (products: ProductType[]) => void;
  selectedProduct?: ProductType;
  updateSelectedProduct: (products: ProductType) => void;
  filterBy: (selectedCategory: string) => void;
  clearFilter: () => void;
};

export type ProductType = {
  id: number;
  title: string;
  price: number;
  description: string;
  category: string;
  image: string;
  matchesSearch: boolean;
};
