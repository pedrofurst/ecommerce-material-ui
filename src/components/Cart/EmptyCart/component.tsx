import { Typography } from '@material-ui/core';
import useStyles from './styles';

type EmptyCartPropsType = {};

function EmptyCart(props: EmptyCartPropsType) {
  const classes = useStyles();
  return (
    <div className={classes.emptyCartContainer}>
      <Typography className={classes.emptyCartText}>
        Your cart is empty.
      </Typography>
    </div>
  );
}

export default EmptyCart;
