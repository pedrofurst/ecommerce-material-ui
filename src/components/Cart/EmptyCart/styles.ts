import { makeStyles } from '@material-ui/core/styles';

export default makeStyles({
  emptyCartContainer: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100%',
  },
  emptyCartText: {
    fontSize: 36,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
