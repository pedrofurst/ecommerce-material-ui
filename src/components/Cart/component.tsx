/* eslint-disable no-debugger */
import {
  Button,
  Divider,
  IconButton,
  Paper,
  Typography,
} from '@material-ui/core';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import React, { useCallback, useMemo } from 'react';
import RemoveShoppingCartIcon from '@material-ui/icons/RemoveShoppingCart';
import { CartType } from '@features/providers/cart/model';
import { ProductType } from '@features/providers/product/model';
import clsx from 'clsx';
import useStyles from './styles';
import EmptyCart from './EmptyCart/component';

type CartPropsType = {
  cart: CartType;
  removeFromCart: (product: ProductType) => void;
  onCheckout: () => void;
  onGoBack: () => void;
  onSelectProduct: (product: ProductType) => void;
};

function Cart(props: CartPropsType) {
  const classes = useStyles();
  const { cart, removeFromCart, onCheckout, onGoBack, onSelectProduct } = props;
  const { products } = cart;

  const total = useMemo(
    () => products.reduce((acc, product) => acc + product.price, 0),
    [products]
  );

  const handleRemoveFromCart = useCallback(
    (product: ProductType) => (e: React.MouseEvent<HTMLElement>) => {
      e.stopPropagation();
      removeFromCart(product);
    },
    [removeFromCart]
  );

  const handleSelectProduct = useCallback(
    (product: ProductType) => () => {
      onSelectProduct(product);
    },
    [onSelectProduct]
  );

  return (
    <Paper className={classes.cartContainer}>
      <IconButton
        role="back-button"
        aria-label="back"
        onClick={onGoBack}
        className={classes.backButton}
      >
        <ArrowBackIcon />
      </IconButton>
      <div className={classes.contentContainer}>
        {products.length === 0 ? (
          <EmptyCart />
        ) : (
          <>
            {products.map((product: ProductType, i: number) => (
              <div key={product.id}>
                <div
                  role="button"
                  tabIndex={product.id}
                  className={clsx(classes.rowContainer, 'product-item')}
                  onClick={handleSelectProduct(product)}
                  onKeyDown={handleSelectProduct(product)}
                >
                  <div className={classes.imageContainer}>
                    <img
                      src={product.image}
                      alt="ImageRow"
                      className={classes.image}
                    />
                  </div>

                  <div className={classes.titleContainer}>
                    <Typography className={classes.title} noWrap>
                      {product.title}
                    </Typography>
                  </div>
                  <div className={classes.priceContainer}>
                    <Typography className={classes.price}>
                      ${product.price.toFixed(2)}
                    </Typography>
                    <IconButton
                      aria-label="remove from shopping cart"
                      onClick={handleRemoveFromCart(product)}
                      className={classes.removeButton}
                      id="remove-from-cart-button"
                      color="secondary"
                    >
                      <RemoveShoppingCartIcon />
                    </IconButton>
                  </div>
                </div>
                {(i !== products.length - 1 || i === 0) && <Divider />}
              </div>
            ))}
            <div className={classes.bottomContainer}>
              <div className={classes.totalRow}>
                <Typography className={classes.total}>
                  {`Total (${products.length} ${
                    products.length === 1 ? 'Item' : 'Items'
                  }):`}
                </Typography>
                <Typography className={classes.price}>
                  ${total.toFixed(2)}
                </Typography>
              </div>
              <div className={classes.proceedToCheckoutContainer}>
                <Button
                  variant="outlined"
                  color="primary"
                  onClick={onCheckout}
                  id="procced-to-checkout-button"
                >
                  Proceed to checkout
                </Button>
              </div>
            </div>
          </>
        )}
      </div>
    </Paper>
  );
}

export default Cart;
