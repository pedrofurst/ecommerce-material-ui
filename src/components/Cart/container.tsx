import { useCallback } from 'react';
import { useHistory } from 'react-router-dom';
import useCartContext from '@features/providers/cart/useCartContext';
import { ProductType } from '@features/providers/product/model';
import useProductContext from '@features/providers/product/useProductContext';
import Cart from './component';

function CartContainer() {
  const { updateSelectedProduct } = useProductContext();
  const { cart, removeFromCart } = useCartContext();
  const history = useHistory();

  const handleNavigateToCheckout = useCallback(
    () => history.push('/checkout'),
    [history]
  );

  const handleOnBack = useCallback(() => {
    history.goBack();
  }, [history]);

  const handleSelectProduct = useCallback(
    (product: ProductType) => {
      updateSelectedProduct(product);
      history.push(`/product/${product.id}`);
    },
    [history, updateSelectedProduct]
  );

  return (
    <Cart
      cart={cart}
      removeFromCart={removeFromCart}
      onCheckout={handleNavigateToCheckout}
      onGoBack={handleOnBack}
      onSelectProduct={handleSelectProduct}
    />
  );
}

export default CartContainer;
