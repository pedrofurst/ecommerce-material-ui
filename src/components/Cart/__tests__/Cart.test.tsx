import { CartType } from '@features/providers/cart/model';
import { IconButton } from '@material-ui/core';
import { shallow } from 'enzyme';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import renderer from 'react-test-renderer';
import { ProductType } from '@features/providers/product/model';
import Cart from '../component';
import EmptyCart from '../EmptyCart/component';

const PRODUCT: ProductType = {
  id: 1,
  title: 'Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops',
  price: 109.95,
  description:
    'Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday',
  category: "men's clothing",
  image: 'https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg',
  matchesSearch: false,
};

const PRODUCT2 = {
  id: 2,
  title: 'Mens Casual Premium Slim Fit T-Shirts ',
  price: 22.3,
  description:
    'Slim-fitting style, contrast raglan long sleeve, three-button henley placket, light weight & soft fabric for breathable and comfortable wearing. And Solid stitched shirts with round neck made for durability and a great fit for casual fashion wear and diehard baseball fans. The Henley style round neckline includes a three-button placket.',
  category: "men's clothing",
  image:
    'https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg',
  matchesSearch: false,
};

describe('<Cart />', () => {
  it('Should show empty card component when there is no products in the cart', () => {
    const cart: CartType = {
      products: [],
    };
    const wrapper = shallow(
      <Cart
        onCheckout={jest.fn()}
        onGoBack={jest.fn()}
        onSelectProduct={jest.fn()}
        cart={cart}
        removeFromCart={jest.fn()}
      />
    );
    expect(wrapper.find(EmptyCart).exists()).toBeTruthy();
  });

  it('Should have back button', () => {
    const wrapper = shallow(
      <Cart
        onCheckout={jest.fn()}
        onGoBack={jest.fn()}
        onSelectProduct={jest.fn()}
        cart={{
          products: [],
        }}
        removeFromCart={jest.fn()}
      />
    );
    expect(wrapper.find(IconButton).exists()).toBeTruthy();
    expect(wrapper.find(ArrowBackIcon).exists()).toBeTruthy();
  });

  it('Should call onGoBack function when the user clicks on back button', () => {
    const onGoBack = jest.fn();
    const wrapper = shallow(
      <Cart
        onCheckout={jest.fn()}
        onGoBack={onGoBack}
        onSelectProduct={jest.fn()}
        cart={{
          products: [],
        }}
        removeFromCart={jest.fn()}
      />
    );
    wrapper.find(IconButton).simulate('click');
    expect(onGoBack).toHaveBeenCalled();
  });

  it('Should call onCheckout when there are products in the cart and the user clicks on Proceed to checkout', () => {
    const onCheckout = jest.fn();
    const wrapper = shallow(
      <Cart
        onCheckout={onCheckout}
        onGoBack={jest.fn()}
        onSelectProduct={jest.fn()}
        cart={{
          products: [PRODUCT],
        }}
        removeFromCart={jest.fn()}
      />
    );
    wrapper.find('#procced-to-checkout-button').simulate('click');
    expect(onCheckout).toHaveBeenCalled();
  });

  it('Should call onSelectProduct when the user selects a product in the cart', () => {
    const onSelectProduct = jest.fn();
    const wrapper = shallow(
      <Cart
        onCheckout={jest.fn()}
        onGoBack={jest.fn()}
        onSelectProduct={onSelectProduct}
        cart={{
          products: [PRODUCT],
        }}
        removeFromCart={jest.fn()}
      />
    );
    wrapper.find('.product-item').simulate('click');
    expect(onSelectProduct).toHaveBeenCalledWith(PRODUCT);
  });

  it('Should call removeFromCart when the user selects a product in the cart', () => {
    const removeFromCart = jest.fn();
    const stopPropagation = jest.fn();
    const wrapper = shallow(
      <Cart
        onCheckout={jest.fn()}
        onGoBack={jest.fn()}
        onSelectProduct={jest.fn()}
        cart={{
          products: [PRODUCT],
        }}
        removeFromCart={removeFromCart}
      />
    );
    wrapper.find('#remove-from-cart-button').simulate('click', {
      stopPropagation,
    });
    expect(removeFromCart).toHaveBeenCalledWith(PRODUCT);
    expect(stopPropagation).toHaveBeenCalled();
  });

  it('Should render correctly when there is one product in the cart ', () => {
    const component = renderer.create(
      <Cart
        onCheckout={jest.fn()}
        onGoBack={jest.fn()}
        onSelectProduct={jest.fn()}
        cart={{
          products: [PRODUCT],
        }}
        removeFromCart={jest.fn()}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Should render correctly when there are more than one product in the cart ', () => {
    const component = renderer.create(
      <Cart
        onCheckout={jest.fn()}
        onGoBack={jest.fn()}
        onSelectProduct={jest.fn()}
        cart={{
          products: [PRODUCT, PRODUCT2],
        }}
        removeFromCart={jest.fn()}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
