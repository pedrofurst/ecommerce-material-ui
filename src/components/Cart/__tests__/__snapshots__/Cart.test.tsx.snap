// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart /> Should render correctly when there are more than one product in the cart  1`] = `
<div
  className="MuiPaper-root makeStyles-cartContainer-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    aria-label="back"
    className="MuiButtonBase-root MuiIconButton-root makeStyles-backButton-2"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="back-button"
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <div
    className="makeStyles-contentContainer-3"
  >
    <div>
      <div
        className="makeStyles-rowContainer-4 product-item"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={1}
      >
        <div
          className="makeStyles-imageContainer-5"
        >
          <img
            alt="ImageRow"
            className="makeStyles-image-6"
            src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
          />
        </div>
        <div
          className="makeStyles-titleContainer-7"
        >
          <p
            className="MuiTypography-root makeStyles-title-8 MuiTypography-body1 MuiTypography-noWrap"
          >
            Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
          </p>
        </div>
        <div
          className="makeStyles-priceContainer-9"
        >
          <p
            className="MuiTypography-root makeStyles-price-10 MuiTypography-body1"
          >
            $
            109.95
          </p>
          <button
            aria-label="remove from shopping cart"
            className="MuiButtonBase-root MuiIconButton-root makeStyles-removeButton-11 MuiIconButton-colorSecondary"
            disabled={false}
            id="remove-from-cart-button"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22.73 22.73L2.77 2.77 2 2l-.73-.73L0 2.54l4.39 4.39 2.21 4.66-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h7.46l1.38 1.38c-.5.36-.83.95-.83 1.62 0 1.1.89 2 1.99 2 .67 0 1.26-.33 1.62-.84L21.46 24l1.27-1.27zM7.42 15c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h2.36l2 2H7.42zm8.13-2c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H6.54l9.01 9zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <hr
        className="MuiDivider-root"
      />
    </div>
    <div>
      <div
        className="makeStyles-rowContainer-4 product-item"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={2}
      >
        <div
          className="makeStyles-imageContainer-5"
        >
          <img
            alt="ImageRow"
            className="makeStyles-image-6"
            src="https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg"
          />
        </div>
        <div
          className="makeStyles-titleContainer-7"
        >
          <p
            className="MuiTypography-root makeStyles-title-8 MuiTypography-body1 MuiTypography-noWrap"
          >
            Mens Casual Premium Slim Fit T-Shirts 
          </p>
        </div>
        <div
          className="makeStyles-priceContainer-9"
        >
          <p
            className="MuiTypography-root makeStyles-price-10 MuiTypography-body1"
          >
            $
            22.30
          </p>
          <button
            aria-label="remove from shopping cart"
            className="MuiButtonBase-root MuiIconButton-root makeStyles-removeButton-11 MuiIconButton-colorSecondary"
            disabled={false}
            id="remove-from-cart-button"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22.73 22.73L2.77 2.77 2 2l-.73-.73L0 2.54l4.39 4.39 2.21 4.66-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h7.46l1.38 1.38c-.5.36-.83.95-.83 1.62 0 1.1.89 2 1.99 2 .67 0 1.26-.33 1.62-.84L21.46 24l1.27-1.27zM7.42 15c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h2.36l2 2H7.42zm8.13-2c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H6.54l9.01 9zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="makeStyles-bottomContainer-15"
    >
      <div
        className="makeStyles-totalRow-14"
      >
        <p
          className="MuiTypography-root makeStyles-total-12 MuiTypography-body1"
        >
          Total (2 Items):
        </p>
        <p
          className="MuiTypography-root makeStyles-price-10 MuiTypography-body1"
        >
          $
          132.25
        </p>
      </div>
      <div
        className="makeStyles-proceedToCheckoutContainer-13"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          disabled={false}
          id="procced-to-checkout-button"
          onBlur={[Function]}
          onClick={[MockFunction]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Proceed to checkout
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Cart /> Should render correctly when there is one product in the cart  1`] = `
<div
  className="MuiPaper-root makeStyles-cartContainer-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    aria-label="back"
    className="MuiButtonBase-root MuiIconButton-root makeStyles-backButton-2"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="back-button"
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </span>
  </button>
  <div
    className="makeStyles-contentContainer-3"
  >
    <div>
      <div
        className="makeStyles-rowContainer-4 product-item"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={1}
      >
        <div
          className="makeStyles-imageContainer-5"
        >
          <img
            alt="ImageRow"
            className="makeStyles-image-6"
            src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
          />
        </div>
        <div
          className="makeStyles-titleContainer-7"
        >
          <p
            className="MuiTypography-root makeStyles-title-8 MuiTypography-body1 MuiTypography-noWrap"
          >
            Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
          </p>
        </div>
        <div
          className="makeStyles-priceContainer-9"
        >
          <p
            className="MuiTypography-root makeStyles-price-10 MuiTypography-body1"
          >
            $
            109.95
          </p>
          <button
            aria-label="remove from shopping cart"
            className="MuiButtonBase-root MuiIconButton-root makeStyles-removeButton-11 MuiIconButton-colorSecondary"
            disabled={false}
            id="remove-from-cart-button"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22.73 22.73L2.77 2.77 2 2l-.73-.73L0 2.54l4.39 4.39 2.21 4.66-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h7.46l1.38 1.38c-.5.36-.83.95-.83 1.62 0 1.1.89 2 1.99 2 .67 0 1.26-.33 1.62-.84L21.46 24l1.27-1.27zM7.42 15c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h2.36l2 2H7.42zm8.13-2c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H6.54l9.01 9zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <hr
        className="MuiDivider-root"
      />
    </div>
    <div
      className="makeStyles-bottomContainer-15"
    >
      <div
        className="makeStyles-totalRow-14"
      >
        <p
          className="MuiTypography-root makeStyles-total-12 MuiTypography-body1"
        >
          Total (1 Item):
        </p>
        <p
          className="MuiTypography-root makeStyles-price-10 MuiTypography-body1"
        >
          $
          109.95
        </p>
      </div>
      <div
        className="makeStyles-proceedToCheckoutContainer-13"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          disabled={false}
          id="procced-to-checkout-button"
          onBlur={[Function]}
          onClick={[MockFunction]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Proceed to checkout
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
